pilot/proxy/envoy/config.go

```diff
@@ -792,6 +792,15 @@ func appendPortToDomains(domains []string, port int) []string {
 	return domainsWithPorts
 }
 
+func truncateClusterName(name string) string {
+	if len(name) > MaxClusterNameLength {
+		prefix := name[:MaxClusterNameLength - sha1.Size * 2]
```

Comment:
why multiply by a factor of 2?

Reply:
sha1.Size is in bytes. `fmt.Printf("%x", ...)`  encodes data to base16 so there are two chars per byte.